### PR TITLE
vite change to api gateway

### PR DESCRIPTION
**Pull request by Ayhanemoon**

### CODE REVIEW

File: src/api/models/exam.js [hunk 1]
```diff
@@ -425,4 +428,52 @@ export default class ExamAPI extends APIRepository {
...
+    })
+  }
+
+  updateReportOptions(data = {}) {
+    return this.sendRequest({
+      apiMethod: 'post',
+      api: this.api,
+      request: this.APIAdresses.report.updateReportOptions(data.examId),
+      resolveCallback: (response) => {
+        return response
```
Comment (kerasus) [inline src/api/models/exam.js:454]: normalize response

File: src/api/models/product.js [hunk 2]
```diff
@@ -45,7 +45,12 @@ export default class ProductAPI extends APIRepository {
       gifts: (id) => this.name + this.APIAdresses.gifts(id),
       sampleContent: (id) => this.name + this.APIAdresses.sampleContent(id),
       categories: this.name + this.APIAdresses.categories,
-      edit: this.name + this.APIAdresses.edit
+      edit: this.name + this.APIAdresses.edit,
+      landing: {
+        sea: {
```
Comment (kerasus) [inline src/api/models/product.js:50]: sea?

File: src/api/models/question.js [hunk 3]
```diff
@@ -212,4 +212,18 @@ export default class QuestionAPI extends APIRepository {
...
     })
   }
+
+  getLevels(data = {}, cache) {
+    return this.sendRequest({
+      apiMethod: 'get',
+      api: this.api,
+      request: this.APIAdresses.levels,
+      resolveCallback: (response) => {
+        return response.data.data
```
Comment (kerasus) [inline src/api/models/question.js:222]: normalize response

File: src/components/Widgets/Quiz/Quiz.vue [hunk 4]
```diff
@@ -192,7 +192,7 @@ export default defineComponent({
         page: 1
       })
         .then((response) => {
-          this.allExamsList = response.ExamList
+          this.allExamsList = response.examList
```
Comment (kerasus) [inline src/components/Widgets/Quiz/Quiz.vue:195]: don't use response

**Review by kerasus — CHANGES_REQUESTED** [review]

File: src/api/models/product.js (hunk 2 above)
Comment (Ayhanemoon) [reply src/api/models/product.js:50]: changed to landingAllProducts

File: src/api/models/question.js [hunk 5]
```diff
@@ -212,4 +212,18 @@ export default class QuestionAPI extends APIRepository {
...
     })
   }
+
+  getLevels(data = {}, cache) {
+    return this.sendRequest({
+      apiMethod: 'get',
+      api: this.api,
+      request: this.APIAdresses.levels,
+      resolveCallback: (response) => {
+        return response.data.data // List of Options Objects [{key,Value}]
```
Comment (kerasus) [inline src/api/models/question.js:222]: create option model

{
    "id": "62f87df6e47ebedac3000d96",
    "type": "reference_type",
    "value": "کنکور",
    "image": null,
    "order": null,
    "updated_at": "2022-08-14 09:15:42",
    "created_at": "2022-08-14 09:15:42"
}

**Review by kerasus — CHANGES_REQUESTED** [review]

File: src/api/models/question.js (hunk 5 above)
Comment (Ayhanemoon) [reply src/api/models/question.js]: this options are not compatible with Option Model

File: src/api/models/exam.js [hunk 6]
```diff
@@ -461,6 +495,36 @@ export default class ExamAPI extends APIRepository {
     })
   }
 
+  konkurTakhminRotbe(data = {}) {
+    return this.sendRequest({
+      apiMethod: 'post',
+      api: this.api,
+      request: this.APIAdresses.konkurTakhminRotbe(data.examId),
+      resolveCallback: (response) => {
+        return response.data.ranks
```
Comment (kerasus) [inline src/api/models/exam.js:504]: normalize response

File: src/api/models/exam.js [hunk 7]
```diff
@@ -252,6 +256,36 @@ export default class ExamAPI extends APIRepository {
...
+        return response.data.data
+      },
+      rejectCallback: (error) => {
+        return error
+      },
+      ...(data.params && { data: data.params })
+    })
+  }
+
+  userQuestionsWithAnswer(data = {}, cache) {
```
Comment (kerasus) [inline src/api/models/exam.js:274]: remove cache

File: src/api/models/exam.js [hunk 8]
```diff
@@ -252,6 +256,36 @@ export default class ExamAPI extends APIRepository {
     })
   }
 
+  userExamInfo(data = {}, cache) {
```
Comment (kerasus) [inline src/api/models/exam.js:259]: default cache

File: src/api/models/exam.js [hunk 9]
```diff
@@ -252,6 +256,36 @@ export default class ExamAPI extends APIRepository {
...
+        return new QuestionList(response.data.data)
+      },
+      rejectCallback: (error) => {
+        return error
+      },
+      data: data.data
+    })
+  }
+
   getAllAnswerOfUser(userExamId, cache) {
```
Comment (kerasus) [inline src/api/models/exam.js:289]: default cache

**Review by kerasus — CHANGES_REQUESTED** [review]

File: src/api/models/exam.js [hunk 10]
```diff
@@ -252,11 +264,45 @@ export default class ExamAPI extends APIRepository {
...
-  getAllAnswerOfUser(userExamId, cache) {
+  userExamInfo(data = {}, cache = { TTL: 100 }) {
+    return this.sendRequest({
+      apiMethod: 'get',
+      api: this.api,
+      request: this.APIAdresses.user.examInfo(data.examId),
+      cacheKey: this.CacheList.user.examInfo(data.examId),
+      ...(cache && { cache }),
+      resolveCallback: (response) => {
+        return response.data.data // Exam Info
```
Comment (kerasus) [inline src/api/models/exam.js:275]: normalize response

File: src/api/models/exam.js [hunk 11]
```diff
@@ -252,11 +264,45 @@ export default class ExamAPI extends APIRepository {
...
+
+  getAllAnswerOfUser(userExamId, cache = { TTL: 100 }) {
     return this.sendRequest({
       apiMethod: 'get',
       api: this.api,
       request: this.APIAdresses.getAllAnswerOfUser(userExamId),
+      cacheKey: this.CacheList.user.getAllAnswerOfUser(userExamId),
+      ...(cache && { cache }),
       resolveCallback: (response) => {
         return response
```
Comment (kerasus) [inline src/api/models/exam.js:307]: normalize response

File: src/api/models/exam.js [hunk 12]
```diff
@@ -266,11 +312,13 @@ export default class ExamAPI extends APIRepository {
...
-  examUserAfterExam(userExamId, cache) {
+  examUserAfterExam(userExamId, cache = { TTL: 100 }) {
     return this.sendRequest({
       apiMethod: 'get',
       api: this.api,
       request: this.APIAdresses.examUserAfterExam(userExamId),
+      cacheKey: this.CacheList.user.examUserAfterExam(userExamId),
+      ...(cache && { cache }),
       resolveCallback: (response) => {
         return response
```
Comment (kerasus) [inline src/api/models/exam.js:323]: normalize response

File: src/api/models/exam.js [hunk 13]
```diff
@@ -310,13 +358,11 @@ export default class ExamAPI extends APIRepository {
...
-  generateExamFile(data = {}, cache) {
+  generateExamFile(data = {}) {
     return this.sendRequest({
       apiMethod: 'post',
       api: this.api,
       request: this.APIAdresses.generateExamFile(data.examId, data.withAnswer),
-      cacheKey: this.CacheList.generateExamFile(data.examId, data.withAnswer),
-      ...(cache && { cache }),
       resolveCallback: (response) => {
         return response // String
```
Comment (kerasus) [inline src/api/models/exam.js:367]: normalize response

File: src/api/models/exam.js [hunk 14]
```diff
@@ -461,6 +507,36 @@ export default class ExamAPI extends APIRepository {
     })
   }
 
+  konkurTakhminRotbe(data = {}) {
+    return this.sendRequest({
+      apiMethod: 'post',
+      api: this.api,
+      request: this.APIAdresses.konkurTakhminRotbe(data.examId),
+      resolveCallback: (response) => {
+        return response.data.ranks // String
```
Comment (kerasus) [inline src/api/models/exam.js:516]: normalize response

File: src/api/models/exam.js [hunk 15]
```diff
@@ -461,6 +507,36 @@ export default class ExamAPI extends APIRepository {
...
+    })
+  }
+
+  takhminRotbe(data = {}) {
+    return this.sendRequest({
+      apiMethod: 'post',
+      api: this.api,
+      request: this.APIAdresses.takhminRotbe,
+      resolveCallback: (response) => {
+        return response.data // object that contains main&sub_category&Zirgorooh
```
Comment (kerasus) [inline src/api/models/exam.js:531]: normalize response

File: src/api/models/exam.js [hunk 16]
```diff
@@ -476,4 +552,20 @@ export default class ExamAPI extends APIRepository {
...
+
+  getExamAnswersFiles(examId, cache = { TTL: 100 }) {
+    return this.sendRequest({
+      apiMethod: 'get',
+      api: this.api,
+      request: this.APIAdresses.pdf(examId),
+      cacheKey: this.CacheList.pdf(examId),
+      ...(cache && { cache }),
+      resolveCallback: (response) => {
+        return response.data // file list
```
Comment (kerasus) [inline src/api/models/exam.js:564]: normalize response

**Review by kerasus — CHANGES_REQUESTED** [review]

File: src/api/models/question.js [hunk 17]
```diff
@@ -198,6 +198,21 @@ export default class QuestionAPI extends APIRepository {
     })
   }
 
+  detach(data) {
+    return this.sendRequest({
+      apiMethod: 'post',
+      api: this.api,
+      request: this.APIAdresses.detach(data.questionId),
+      resolveCallback: (response) => {
+        return response.data
```
Comment (kerasus) [inline src/api/models/question.js:207]: normalize response

**Review by kerasus — CHANGES_REQUESTED** [review]

**Review by kerasus — APPROVED** [review]

**Pull request merged**